Description: The version of botocore in Ubuntu has removed
  the vendored version of urllib exceptions. Switch to the
  exception class in botocore.exceptions.
Author: Corey Bryant
Forwarded: no

--- a/cinder/backup/drivers/s3.py
+++ b/cinder/backup/drivers/s3.py
@@ -74,8 +74,6 @@
 import boto3
 from botocore.config import Config
 from botocore import exceptions as boto_exc
-from botocore.vendored.requests.packages.urllib3 import exceptions as \
-    urrlib_exc
 from oslo_config import cfg
 from oslo_log import log as logging
 from oslo_utils.secretutils import md5
@@ -369,7 +367,7 @@
                 resp = self.conn.get_object(**get_args)
                 return resp.get('Body').read()
             except (socket.timeout, socket.error,
-                    urrlib_exc.ReadTimeoutError,
+                    boto_exc.ReadTimeoutError,
                     boto_exc.IncompleteReadError) as e:
                 last_exception = e
                 continue
